eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,38 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper($servicePrincipal)
+{
+    $spPassword = ""
+    if ($servicePrincipal.GetType().Name -eq 'MicrosoftGraphServicePrincipal') {
+        Write-Host "Creating password for service principal via MS Graph API"
+        # Microsoft graph objects (Az version >= 7.0.0) do not provision a secret # on creation so it must be added separately.
+        # Submitting a password credential object without specifying a password will result in one being generated on the server side.
+        $password = New-Object -TypeName "Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.MicrosoftGraphPasswordCredential"
+        $password.DisplayName = "Password for $displayName"
+        $credential = Retry { New-AzADSpCredential -PasswordCredentials $password -ServicePrincipalObject $servicePrincipal }
+        $spPassword = ConvertTo-SecureString $credential.SecretText -AsPlainText -Force
+    } else {
+        # Secret property exists on PSADServicePrincipal type from AAD graph in Az # module versions < 7.0.0
+        $spPassword = $servicePrincipal.Secret
+    }
+
+    # MicrosoftGraphServicePrincipal objects have AppId, AAD graph objects have ApplicationId
+    $appId = $servicePrincipal.AppId ? $servicePrincipal.AppId : $servicePrincipal.ApplicationId
```

Comment:
What if there is an `AppId` property but it happens to be `$null` and there is no `ApplicationId` property?    Should we consider `Get-Member` here to probe and make sure that one or the other exists?